### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by galmalka122**

### CODE REVIEW

File: pptx_clarifier/__init__.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import asyncio
+import os
+import sys
+import threading
```
Comment (excellenteam-scaleup) [inline pptx_clarifier/__init__.py:4]: why do you work with threads?
the heavy long part is the GPT part, which is dealt using asyncio

File: pptx_clarifier/client/Status.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import datetime
+
+
+class Status:
```
Comment (excellenteam-scaleup) [inline pptx_clarifier/client/Status.py:4]: Good

File: pptx_clarifier/definitions.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import os
+import sys
+
+root = sys.path[1]
+uploads_directory = os.path.join(root, "pptx_clarifier", "uploads")
+outputs_directory = os.path.join(root, "pptx_clarifier", "outputs")
+os.makedirs(uploads_directory, exist_ok=True)
```
Comment (excellenteam-scaleup) [inline pptx_clarifier/definitions.py:7]: Nice flag

File: pptx_clarifier/definitions.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+os.makedirs(outputs_directory, exist_ok=True)
+
+base_url = 'localhost:5000'
```
Comment (excellenteam-scaleup) [inline pptx_clarifier/definitions.py:10]: didn't I already see that somewhere?

File: pptx_clarifier/logger.py
Comment (excellenteam-scaleup) [inline pptx_clarifier/logger.py:1]: Nice

File: pptx_clarifier/pptx_clarifier_api/web_api.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    """
+
+    uploaded_file = request.files['file']
+    file_uuid = process_file(uploaded_file)
+    logger.info(f"file uploaded with uid: {file_uuid}")
+    return make_response({"file_uuid": file_uuid, "message": "file uploaded successfully"}, 200)
+
+
+@app.get('/upload')
+def upload_file():
```
Comment (excellenteam-scaleup) [inline pptx_clarifier/pptx_clarifier_api/web_api.py:72]: what is this for?

File: pptx_clarifier/pptx_explainer/__init__.py
Comment (excellenteam-scaleup) [inline pptx_clarifier/pptx_explainer/__init__.py:1]: Not sure the init file is where you want to save your prompt

File: pptx_clarifier/pptx_explainer/__init__.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+explainer_logger = logger.setup_log("clarifier", __name__)
+
+initial_prompt = """
+You supply concise and to the point explanations about PowerPoint slides.
+
+
+###
+
+
+Instructions: 1. Please wait for the first slide before generating explanations. The slide data will be provided
```
Comment (excellenteam-scaleup) [inline pptx_clarifier/pptx_explainer/__init__.py:12]: Interesting approach.
Does this prompt synchronized with the demand to run it a-synchronically? we don't want GPT to have dependencies between slides which will make it wait and take longer..
I might give GPT more context and less instructions.

File: pptx_clarifier/pptx_explainer/explainer.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    files_in_process.remove(get_file_name(file_name))
+
+
+async def clarify_file(file_name: str):
+    # Add the file to the files in process set
+    files_in_process.add(get_file_name(file_name))
+    logger.info(f"Processing {file_name}")
+    try:
+        json_path = await clarify(file_name)
+    except FileNotFoundError as file_not_found_error:
```
Comment (excellenteam-scaleup) [inline pptx_clarifier/pptx_explainer/explainer.py]: Nice error handling

File: pptx_clarifier/pptx_explainer/presentation_parser.py [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    if not os.path.isabs(path):
+        path = os.path.abspath(path)
+    if not os.path.exists(path):
+        logger.error('User entered an invalid path')
+        raise FileNotFoundError(f'{path} does not exist')
+    if not path.endswith('.pptx'):
+        raise ValueError(f'{path} is not a pptx file')
+
+
+if __name__ == '__main__':
```
Comment (excellenteam-scaleup) [inline pptx_clarifier/pptx_explainer/presentation_parser.py]: have a main function please

**Review by excellenteam-scaleup — APPROVED** [review]
Good job.
Seems like you put a lot of work into this system.
The code looks good, and well separated into modules and functions.
I think you sometimes have unneeded documentation, and should think whether the documentation really adds information that is not covered by the function's name.
Please also respond to the comment about your prompt, and good luck in the next stage!

File: pptx_clarifier/pptx_explainer/__init__.py (hunk 6 above)
Comment (galmalka122) [reply pptx_clarifier/pptx_explainer/__init__.py:12]: hi :)
though the gpt responses are a-sync the requests being sent synchronize (each slide is sent to gpt and while the program waits to the response, it sends another slides, but in order), so the prompt being sent for each slide includes the previous slides in it.

File: pptx_clarifier/pptx_clarifier_api/web_api.py (hunk 5 above)
Comment (galmalka122) [reply pptx_clarifier/pptx_clarifier_api/web_api.py:72]: it is for testing purposes which i forgot to delete :/ , and discovered posting file using postman right after :|

File: pptx_clarifier/__init__.py (hunk 1 above)
Comment (galmalka122) [reply pptx_clarifier/__init__.py:4]: i couldn't figure out how to run both clarifier and the api

**Review by excellenteam-scaleup — APPROVED** [review]
Better to have new PR for each exercise, but good fixes to the comments.

**Pull request merged**